chia/full_node/full_node.py

```diff
@@ -787,8 +797,18 @@ async def receive_block_batch(
                     self.log.error(f"Error: {error}, Invalid block from peer: {peer.get_peer_info()} ")
                 return False, advanced_peak, fork_height
             block_record = self.blockchain.block_record(block.header_hash)
-            if block_record.sub_epoch_summary_included is not None and self.weight_proof_handler is not None:
-                await self.weight_proof_handler.create_prev_sub_epoch_segments()
+            if block_record.sub_epoch_summary_included is not None:
+                if wp_summaries is not None:
+                    # make sure this matches the summary list we got
+                    idx = len(self.blockchain.get_ses_heights())
```

Comment:
We should make sure we are testing reorgs where the peak is much higher than the fork point (deep reorg). 
Also need to test with mainnet

Reply:
yes im going to change this to check it inside receive_block it should happen before we add the new sub epoch summary, it would also handle reorgs better